lib/core/cpuMonitor.cpp

```diff
@@ -0,0 +1,172 @@
+#include "cpuMonitor.hpp"
+
+#include "kotekanLogging.hpp" // for WARN_NON_OO
+
+#include "json.hpp" // for json, basic_json<>::object_t, basic_json<>::value_type
+
+#include <chrono>     // for operator""ms, chrono_literals
+#include <exception>  // for exception
+#include <functional> // for _Bind_helper<>::type, _Placeholder, bind, _1, placeholders
+#include <math.h>     // for floor
+#include <pthread.h>  // for pthread_setaffinity_np
+#include <sched.h>    // for cpu_set_t, CPU_SET, CPU_ZERO
+#include <stdio.h>    // for fclose, fopen, fscanf, snprintf, FILE
+#include <unistd.h>   // for pid_t, sysconf, _SC_NPROCESSORS_ONLN
+#include <utility>    // for pair
+#include <vector>     // for vector
+
+using namespace std::chrono_literals;
+using namespace std::placeholders;
+
+namespace kotekan {
+
+CpuMonitor::CpuMonitor() {
+    // Register CPU usage callback
+    restServer::instance().register_get_callback(
+        "/cpu_ult", std::bind(&CpuMonitor::cpu_ult_call_back, this, _1));
+}
+
+CpuMonitor::~CpuMonitor() {
+    restServer::instance().remove_get_callback("/cpu_ult");
+
+    try {
+        this_thread.join();
```

Comment:
You should also check if we ever started the CPU Monitor, no need to have a warning if we never intended this thread to be joinable.    And it might be a good idea to call `stop()` inside this function just in case it isn't called externally.